src/main/java/dev/braintrust/trace/Attachment.java

```diff
@@ -0,0 +1,121 @@
+package dev.braintrust.trace;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Base64;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+
+/**
+ * Utility to serialize LLM attachment data in a braintrust-friendly manner.
+ *
+ * <p>Users of the SDK likely don't need to use this utility directly because instrumentation will
+ * properly serialize messages out of the box.
+ *
+ * <p>The serialized json will conform to the otel input/output GenericPart schema. See
+ * https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-input-messages.json and
+ * https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-output-messages.json
+ */
+public class Attachment {
```

Comment:
maybe start out with this as `Base64Attachment`? we want to eventually support `braintrust_attachment` types which are called `Attachment` in the py/ts sdks